api/src/main/java/org/pmiops/workbench/access/AccessTierServiceImpl.java

```diff
@@ -18,16 +23,126 @@
+      Provider<WorkbenchConfig> configProvider,
+      AccessTierDao accessTierDao,
+      UserAccessTierDao userAccessTierDao) {
     this.configProvider = configProvider;
     this.accessTierDao = accessTierDao;
+    this.userAccessTierDao = userAccessTierDao;
   }
 
   /**
-   * A placeholder implementation until we complete the Controlled Tier access modules.
+   * Return all access tiers in the database
+   *
+   * @return the List of all DbAccessTiers in the database
+   */
+  public List<DbAccessTier> getAllTiers() {
+    return accessTierDao.findAll();
+  }
+
+  /**
+   * Return the access tier referred to by the shortName in the database
+   *
+   * @param shortName the short name of the access tier to look up in the database
+   * @return an {@code Option<DbAccessTier>} if one matches the shortName passed in, EMPTY otherwise
+   */
+  public Optional<DbAccessTier> getAccessTier(String shortName) {
+    return accessTierDao.findOneByShortName(shortName);
+  }
+
+  /**
+   * Return the Registered Tier if it exists in the database
+   *
+   * @return a DbAccessTier representing the Registered Tier
+   * @throws ServerErrorException if there is no Registered Tier
+   */
+  public DbAccessTier getRegisteredTier() {
+    return getAccessTier(REGISTERED_TIER_SHORT_NAME)
+        .orElseThrow(() -> new ServerErrorException("Cannot find Registered Tier in database."));
+  }
+
+  /**
+   * Add memberships to all tiers for a user if they don't exist by inserting DB row(s). For any
+   * memberships which exist, update them (whether enabled or not) set them to ENABLED.
+   *
+   * @param user the DbUser in the user-accessTier mappings we're updating
+   */
+  public void addUserToAllTiers(DbUser user) {
+    getAllTiers().forEach(tier -> addUserToTier(user, tier));
+  }
+
+  /**
+   * Add a Registered Tier membership to a user if none exists by inserting a DB row. If such a
+   * membership exists (whether enabled or not) set it to ENABLED.
+   *
+   * @param user the DbUser in the user-accessTier mapping we're updating
+   */
+  public void addUserToRegisteredTier(DbUser user) {
+    addUserToTier(user, getRegisteredTier());
+  }
+
+  /**
+   * Remove a Registered Tier membership from a user if one exists by marking that membership as
+   * DISABLED. Do nothing if no membership exists.
+   *
+   * @param user the DbUser in the user-accessTier mapping we're updating
+   */
+  public void removeUserFromRegisteredTier(DbUser user) {
+    removeUserFromTier(user, getRegisteredTier());
+  }
+
+  /**
+   * Add a tier membership to a user if none exists by inserting a DB row. If such a membership
+   * exists (whether enabled or not) set it to ENABLED.
+   *
+   * @param user the DbUser in the user-accessTier mapping we're updating
+   * @param accessTier the DbAccessTier in the user-accessTier mapping we're updating
+   */
+  private void addUserToTier(DbUser user, DbAccessTier accessTier) {
```

Comment:
Given the name of this method, I would have also expected this to synchronize the user's membership within the auth domain group. Will it do that after the migration? If not - where will this happen? (possibly method doc should indicate this)

Reply:
I'll add a comment explaining that group sync remains part of add/remove...Idempotent for now.  Yes, AccessTierService will handle this post-migration.